server/functions/index.js

```diff
@@ -3,31 +3,73 @@ const functions = require('firebase-functions');
 
 const admin = require('firebase-admin');
 admin.initializeApp();
+var db = admin.firestore();
 
-exports.sendNotificationOnDBChange = functions.firestore
-    .document('favors/{favorId}')
-    .onWrite((change, context) => {
+// function to calculate the distance between two points given their coordinates
+function distanceInKm(lon1, lat1, lon2, lat2) {
+    var R = 6371; // Radius of the earth in km
+    var dLat = (lat2 - lat1) * Math.PI / 180;  // Javascript functions in radians
+    var dLon = (lon2 - lon1) * Math.PI / 180;
+    var a = Math.sin(dLat / 2) * Math.sin(dLat / 2) +
+        Math.cos(lat1 * Math.PI / 180) * Math.cos(lat2 * Math.PI / 180) *
+        Math.sin(dLon / 2) * Math.sin(dLon / 2);
+    var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
+    return R * c;
+}
 
-        // This registration token comes from the client FCM SDKs.
-        var registrationToken = 'TOKEN HERE';
+function sendMessage(message, usersIds) {
+    admin.messaging().sendMulticast(message)
+        .then((response) => {
+            if (response.failureCount > 0) {
+                const failedTokens = [];
+                response.responses.forEach((resp, idx) => {
+                    if (!resp.success) {
+                        failedTokens.push(usersIds[idx]);
+                    }
+                });
+                console.log('List of tokens that caused failures: ' + failedTokens);
+            }
+        });
+}
 
-        var message = {
+// send new favor notification to users in the area
+exports.sendNotificationNearbyOnNewFavor = functions.firestore
+    .document('/favors/{favorId}')
```

Comment:
I'd write the following so it's a bit more readable.
```
firestore
  .collection("favors")
  .document(favorId)
```

Reply:
I'm not really sure it would work the same way, {favorId} is just a placeholder for indicating any favor in the favors collection

Reply:
Yeah you're probably right. I wrote this comment before I realized this was javascript code lol.